### PR TITLE
Add support for the Huawei smartax display version command. Command + Tests

### PR DESCRIPTION
**Pull request by evilmonkey19**
It adds supports as suggested in the issue #1624. This is the first command.

*display version*

### CODE REVIEW

**Comment by evilmonkey19** [comment]
The pull request is ready for the code review! Thanks for the help :)

**Comment by evilmonkey19** [comment]
Okey, now it is already tested on a real Huawei smartax and it works. Sorry for all the commits ✅

File: ntc_templates/templates/huawei_smartax_display_ont_info_by-sn_sn.textfsm [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+Value DBA_TYPE (\w+)
+Value DISTANCE_M (\d+)
+Value LAST_DISTANCE_M (\d+)
+Value BATTERY_STATE (\w+\s*\w*)
+Value POWER_TYPE (\w+)
+Value MEMORY_OCCUPATION (\d+)
+Value CPU_OCCUPATION (\d+)
+Value TEMPERATURE (\d+\(\w\))
+Value AUTHENTIC_TYPE (\S+)
+Value SN (\w+\s*\(*\S*\))
```
Comment (jvanderaa) [inline ntc_templates/templates/huawei_smartax_display_ont_info_by-sn_sn.textfsm:15]:
```suggestion
Value SERIAL (\w+\s*\(*\S*\))
```

File: ntc_templates/templates/huawei_smartax_display_ont_info_fsp.textfsm [hunk 2]
```diff
@@ -0,0 +1,19 @@
+Value Required FSP (\w+\/\s\w+\/\w+)
+Value Key ONT_ID (\d+)
+Value SN (\w+)
```
Comment (jvanderaa) [inline ntc_templates/templates/huawei_smartax_display_ont_info_fsp.textfsm:3]:
```suggestion
Value SERIAL (\w+)
```

File: ntc_templates/templates/huawei_smartax_display_ont_info_summary_sn.textfsm [hunk 3]
```diff
@@ -0,0 +1,17 @@
+Value Key ONT_ID (\d+)
+Value SN (\w+)
```
Comment (jvanderaa) [inline ntc_templates/templates/huawei_smartax_display_ont_info_summary_sn.textfsm:2]:
```suggestion
Value SERIAL (\w+)
```

**Review by jvanderaa — CHANGES_REQUESTED** [review]
Just looking to convert `SN` to `SERIAL` for cross vendor capability.

File: ntc_templates/templates/huawei_smartax_display_ont_info_by-sn_sn.textfsm (hunk 1 above)
Comment (jvanderaa) [inline ntc_templates/templates/huawei_smartax_display_ont_info_by-sn_sn.textfsm:15]:
```suggestion
Value SERIAL_NUMBER (\w+\s*\(*\S*\))
```

**Comment by evilmonkey19** [comment]
> Just looking to convert `SN` to `SERIAL` for cross vendor capability.

Done 👍

**Comment by evilmonkey19** [comment]
There was a really small bug in the tests due to the change of SERIAL_NUMBER. Now it is completely fixed :)

**Comment by evilmonkey19** [comment]
I close this pull-request as I have done new pull-requests spliting the commands as suggested by @jvanderaa. Thanks for the help!

**Pull request closed without merging**